devito/types/basic.py

```diff
@@ -83,6 +83,9 @@ def _C_typedata(self):
         The type of the object in the generated code as a `str`.
         """
         _type = self._C_ctype
+        if isinstance(_type, CustomDtype):
```

Comment:
I genuinely wonder why we haven't needed this all along? 

Though CustomDtype is probably used way more in PRO than in OSS

Reply:
I thought so too